exersices/ex05/bounded_sim.py

```diff
@@ -0,0 +1,91 @@
+# -*- coding: utf-8 -*-
+
+__author__ = 'sebastian kihle'
+__email__ = 'sebaskih@nmbu.no'
+
+from walker_sim import Walker, Simulation
+
+
+class BoundedWalker(Walker):
+    def __init__(self, start, home, left_limit, right_limit):
+        """
+        Initialise the walker
+
+        Arguments
+        ---------
+        start : int
+            The walker's initial position
+        home : int
+            The walk ends when the walker reaches home
+        left_limit : int
+            The left boundary of walker movement
+        right_limit : int
+            The right boundary  of walker movement
+        """
+        super().__init__(start, home)
+        self.start = start
+        self.home = home
+        self.left_limit = left_limit
+        self.right_limit = right_limit
+
+    def single_run(self):
+        while not self.is_at_home:
+            self.move()
+            if self.get_position() < self.left_limit:
+                self.position += 1
+            if self.get_position() > self.right_limit:
+                self.position -= 1
+        return self.get_steps()
+
+
+class BoundedSimulation(Simulation):
+    def __init__(self, start, home, seed, left_limit, right_limit):
+        """
+        Initialise the simulation
+
+        Arguments
+        ---------
+        start : int
+            The walker's initial position
+        home : int
+            The walk ends when the walker reaches home
+        seed : int
+            Random generator seed
+        left_limit : int
+            The left boundary of walker movement
+        right_limit : int
+            The right boundary  of walker movement
+        """
+        super().__init__(start, home, seed)
+        self.start = start
+        self.home = home
+        self.seed = seed
+        self.left_limit = left_limit
+        self.right_limit = right_limit
+
+    def single_walk(self):
+
+        f = BoundedWalker(self.start, self.home, self.right_limit,
+                          self.left_limit)
+        while not f.is_at_home():
```

Comment:
this part of the code is duplicated in both Walker class and the Simulation class, it has to be done here only
also the same thing, you should fix the number of steps here